etcdserver/config.go

```diff
@@ -105,7 +108,12 @@ func (c *ServerConfig) verifyLocalMember(strict bool) error {
 
 func (c *ServerConfig) MemberDir() string { return path.Join(c.DataDir, "member") }
 
-func (c *ServerConfig) WALDir() string { return path.Join(c.MemberDir(), "wal") }
+func (c *ServerConfig) WALDir() string {
```

Comment:
Couldn't this logic be moved into the initialization of the `ServerConfig`? IE: You can do this logic in `startEtcd()` and then just assign to `config.WALDir` (and then remove this method).